### PR TITLE
feat(trace): correct loading and add better error state

### PR DESCRIPTION
**Pull request by JonasBa**
rename GetTraceType to shape as it is more descriptive and more accurate of what it does (the shape can also change through the course as we fetch more data). 

Adds a proper loading and error state based on the request status

### CODE REVIEW

File: static/app/components/feedback/widget/useFeedbackWidget.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import ConfigStore from 'sentry/stores/configStore';
 import {useLegacyStore} from 'sentry/stores/useLegacyStore';
 
 interface Props {
-  buttonRef?: RefObject<HTMLButtonElement>;
+  buttonRef?: RefObject<HTMLButtonElement> | RefObject<HTMLAnchorElement>;
```
Comment (evanpurkhiser) [inline static/app/components/feedback/widget/useFeedbackWidget.tsx:10]:
```suggestion
  buttonRef?: RefObject<HTMLButtonElement | HTMLAnchorElement>;
```

can't it be written like this

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/components/feedback/widget/useFeedbackWidget.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/feedback/widget/useFeedbackWidget.tsx:10]: Oh really? What's the difference in this scenario?

Comment (JonasBa) [reply static/app/components/feedback/widget/useFeedbackWidget.tsx:10]: Yeah, correct. Hopefully this [explains](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3gJRQMwPIBGAVigMYwC+cOUEIcARFCgIbkMBQHOArgHblgEPnADuwACYoAFMxwAuONnzEyMADwAJACoBZADIAhHjBjCAogBsUIFHxgAfHQYCCAgBbQrNuzAB8AJToFNz8gsJwfCxQtKKyuIrKhCTkWnpGJmZ83rb2fnAOSrjJammuHl7Wuf5BIRykwgDO8OJSAPpyicWqqc4ZphZVvk7pbqSeUDm++QC8GHCkPDG+inw8lpYhDXzNkdGxHQlFKikafcYD2UN5cHPoC0vM9qvrm1ytMh+HOAEcUTEQOJfOS-f6xaRgwHfX5cIA) it

Comment (evanpurkhiser) [reply static/app/components/feedback/widget/useFeedbackWidget.tsx:10]: Yeah I see what you're saying. Subtle. 👍

**Pull request merged**